File: src/main/kotlin/algorithms/easy/RemoveElement.kt
package br.com.rbthmn.algorithms.easy

import br.com.rbthmn.Problem
import br.com.rbthmn.algorithms.utils.moveElementToLastIndex

@Problem(27)
fun removeElement(nums: IntArray, `val`: Int): Int {
    var k = 0
    var i =  nums.size - 1
    while (i >= 0) {
        if (nums[i] == `val`) {
                moveElementToLastIndex(nums, i)
        } else {
            k++
        }
        i--
    }

    return k
}

File: src/main/javascript/algorithms/easy/RemoveElement.js
// Problem 27
var removeElement = function(nums, val) {
    let counter = 0
    for (let i = 0; i <= nums.length - 1; i++) {
        if (nums[i] !== val) {
            nums[counter] = nums[i]
            counter++
        }
    }
    return counter
};